Mage/src/main/java/mage/watchers/Watcher.java

```diff
@@ -76,6 +78,15 @@ protected String getBasicKey(){
 
     public abstract void watch(GameEvent event, Game game);
 
-    public abstract Watcher copy();
+    public <T extends Watcher> T copy(){
+        try {
+            Constructor<? extends Watcher> constructor = this.getClass().getDeclaredConstructor(getClass());
+            constructor.setAccessible(true);
+            return (T) constructor.newInstance(this);
+        } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+            e.printStackTrace();
```

Comment:
No needs in catch and hide exceptions -- if class gets error then xmage must known about that critical error (e.g. on tests or in game).

Reply:
So you say that the exceptions need to be thrown up?

Reply:
Yes, remove catch code at all or thrown up exception. There are not use cases for broken copy/classes.